lib/ash_admin/components/resource/form.ex

```diff
@@ -196,7 +196,11 @@ defmodule AshAdmin.Components.Resource.Form do
             class="block text-sm font-medium text-gray-700"
             for={@form.name <> "[#{attribute.name}]"}
           >
-            <%= to_name(attribute.name) %>
+            <%= if foreign_key?(@resource, attribute) do %>
+              <%= attribute.name |> to_string() |> String.replace_suffix("_id", "") |> to_name() %>
```

Comment:
I would say this should turn into the relationship name that it maps to. So like `<%= if relationship = foreign_key_relationship(....)`